NumLib/DOF/DOFTableUtil.h

```diff
@@ -0,0 +1,27 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#ifndef NUMLIB_DOF_DOFTABLEUTIL_H
+#define NUMLIB_DOF_DOFTABLEUTIL_H
+
+#include "NumLib/DOF/LocalToGlobalIndexMap.h"
+
+namespace NumLib
+{
+//! Returns nodal indices for the item identified by \c id from the given
+//!  \c dof_table.
+std::vector<GlobalIndexType> getIndices(
+    std::size_t const id, NumLib::LocalToGlobalIndexMap const& dof_table);
+
+//! Returns the values for the given \c dof_indices from the vector \c x.
+std::vector<double> getLocalNodalDOFs(
```

Comment:
Isn't this a function to create a sub-vector from the given positions? why not have it somewhere in MathLib?